Common/Securities/Futures/FutureMarginModel.cs

```diff
@@ -0,0 +1,345 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+
+using System.Globalization;
+using QuantConnect.Orders;
+using QuantConnect.Logging;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Represents a simple margining model for margining futures. Margin file contains Initial and Maintenance margins 
+    /// </summary>
+    public class FutureMarginModel : ISecurityMarginModel
+    {
+        private static readonly object DataFolderSymbolLock = new object();
+
+        // historical database of margin requirements
+        private MarginRequirementsEntry[] _marginRequirementsHistory;
+        private int _marginCurrentIndex;
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FutureMarginModel"/>
+        /// </summary>
+        public FutureMarginModel()
+        {
+        }
+
+        /// <summary>
+        /// Gets the current leverage of the security
+        /// </summary>
+        /// <param name="security">The security to get leverage for</param>
+        /// <returns>The current leverage in the security</returns>
+        public virtual decimal GetLeverage(Security security)
+        {
+            return 1/GetMaintenanceMarginRequirement(security, security.Holdings.HoldingsCost);
+        }
+
+        /// <summary>
+        /// Sets the leverage for the applicable securities, i.e, futures
+        /// </summary>
+        /// <remarks>
+        /// This is added to maintain backwards compatibility with the old margin/leverage system
+        /// </remarks>
+        /// <param name="security"></param>
+        /// <param name="leverage">The new leverage</param>
+        public virtual void SetLeverage(Security security, decimal leverage)
+        {
+            // Futures are leveraged products and different leverage cannot be set by user.
```

Comment:
is this an `InvalidOperationException` ? If I wrote an algo that set leverage on a future I would want to know that what I'm doing can't be done